src/components/Compendia/FileDownloadForm.js

```diff
@@ -0,0 +1,83 @@
+import { useEffect, useState } from 'react'
+import { Heading, Text } from 'grommet'
+import { useRefinebio } from 'hooks/useRefinebio'
+import { useResponsive } from 'hooks/useResponsive'
+import formatBytes from 'helpers/formatBytes'
+import formatString from 'helpers/formatString'
+import { Anchor } from 'components/shared/Anchor'
+import { Button } from 'components/shared/Button'
+import { CheckBox } from 'components/shared/CheckBox'
+import { Column } from 'components/shared/Column'
+import { Row } from 'components/shared/Row'
+import { links } from 'config'
+
+export const FileDownloadForm = ({ compendium }) => {
+  const { setResponsive } = useResponsive()
+  const { acceptedTerms, setAcceptedTerms } = useRefinebio()
+  const [isTermsChecked, setIsTermsChecked] = useState(acceptedTerms)
+  const [submitted, setSubmitted] = useState(false)
+
+  useEffect(() => {
+    if (submitted) {
+      setAcceptedTerms(isTermsChecked)
+    }
+  }, [acceptedTerms, submitted])
+
+  return (
+    <Column align={setResponsive('center', 'start')}>
+      <Heading level={1} margin={{ bottom: 'small' }}>
+        Download the {formatString(compendium.primary_organism_name)} compendium
+      </Heading>
+      <Text>
+        Download size: {formatBytes(compendium.computed_file.size_in_bytes)}
+      </Text>
+      <Row direction={setResponsive('column', 'column', 'row')} width="100%">
+        <Row
+          align={setResponsive('start', 'start', 'center')}
+          direction={setResponsive('column', 'column', 'row')}
+          justify="start"
+          margin={{
+            top: 'small'
+          }}
+          fill
+        >
+          <Column align={setResponsive('center', 'start')}>
+            <CheckBox
+              label={
```

Comment:
```suggestion
              checked={isTermsChecked}
              label={
```

And then use the event to set `isTermsChecked` based on the state of the checkbox that was derived from the initial local state.